### PR TITLE
fix: [Go] added missing prompt metadata

### PR DESCRIPTION
**Pull request by pavelgj**

### CODE REVIEW

File: go/plugins/dotprompt/dotprompt.go [hunk 1]
```diff
@@ -66,6 +66,9 @@ type Prompt struct {
 	// The template for the prompt.
 	Template *raymond.Template
 
+	// The template for the prompt.
```
Comment (ianlancetaylor) [inline go/plugins/dotprompt/dotprompt.go:69]: This is the same comment as for the `Template` field.  Perhaps `// The parsed template.` and `// The unparsed template.`.

File: go/plugins/dotprompt/genkit.go [hunk 2]
```diff
@@ -52,6 +52,9 @@ func (p *Prompt) buildVariables(variables any) (map[string]any, error) {
 	}
 
 	v := reflect.Indirect(reflect.ValueOf(variables))
+	if v.Kind() == reflect.Map {
+		return variables.(map[string]any), nil
+	}
 	if v.Kind() != reflect.Struct {
 		return nil, errors.New("dotprompt: fields not a struct or pointer to a struct")
```
Comment (ianlancetaylor) [inline go/plugins/dotprompt/genkit.go:59]: "... or a map"

File: go/plugins/dotprompt/genkit.go [hunk 3]
```diff
@@ -137,8 +140,16 @@ func (p *Prompt) Register() error {
 	if p.Variant != "" {
 		name += "." + p.Variant
 	}
-
-	p.action = ai.DefinePrompt("dotprompt", name, nil, p.buildRequest, p.Config.InputSchema)
+	// TODO: Unwind this change once Monaco Editor supports newer than JSON schema draft-07.
+	p.InputSchema.Version = ""
```
Comment (ianlancetaylor) [inline go/plugins/dotprompt/genkit.go:144]: I suggest a blank line before the TODO comment and after this line to clarify what "unwind this change" refers to.

File: go/plugins/dotprompt/genkit.go [hunk 4]
```diff
@@ -137,8 +140,16 @@ func (p *Prompt) Register() error {
...
+	p.action = ai.DefinePrompt("dotprompt", name, map[string]any{
```
Comment (ianlancetaylor) [inline go/plugins/dotprompt/genkit.go:145]: I think the code will be easier to read if written as

```Go
    metadata := map[string]any{
        "prompt": ...
        ...
    }
    p.action = ai.DefinePrompt("dotprompt", name, metadata, p.buildRequest, p.Config.InputSchema)
```

**Review by ianlancetaylor — APPROVED** [review]

File: go/plugins/dotprompt/dotprompt.go (hunk 1 above)
Comment (pavelgj) [reply go/plugins/dotprompt/dotprompt.go]: Good catch! Done.

File: go/plugins/dotprompt/genkit.go (hunk 2 above)
Comment (pavelgj) [reply go/plugins/dotprompt/genkit.go]: Done.

File: go/plugins/dotprompt/genkit.go (hunk 4 above)
Comment (pavelgj) [reply go/plugins/dotprompt/genkit.go]: Done.

File: go/plugins/dotprompt/genkit.go (hunk 3 above)
Comment (pavelgj) [reply go/plugins/dotprompt/genkit.go:145]: Done.

**Pull request merged**